File: En C/exercice3_4.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
int main ()
{
    char choix , rep[255];
    do
    {  
        printf("entrez un repertoire : ");
        scanf ("%s",rep);
        pid_t pid_fils = fork();
        if (pid_fils == 0) execl("/bin/cp", "cp", "-r", ".", rep, NULL);
        else wait(NULL) ;
        printf(" Entrez Q pour quitter\n");
        scanf (" %c" , &choix) ;  
    }
    while ((choix != 'Q')&&(choix!='q'));
    return(0) ;
}

File: en Python/exercice3_4.py
import os
import sys

while True:
    rep = input("entrez un repertoire : ")
    pid = os.fork()
    if pid == 0: 
        os.execl("/bin/cp", "cp", "-r", ".", rep)
    os.wait()
    choix = input("Entrez Q pour quitter : ")
    if choix.lower() == 'q':
        break